<!DOCTYPE html>
<html lang="en">
<head>
    <meta  name="viewport" content="width=device-width, initial-scale=1" charset="UTF-8">
    <title>merlin v4 Tutorial</title>
    <link rel="stylesheet" href="static/css/sidenav.css">
    <link rel="stylesheet" href="static/css/sticky.css">
    <link rel="stylesheet" href="static/css/accordion.css">
    <link rel="stylesheet" href="static/css/panel.css">
    <link rel="stylesheet" href="static/css/tutorialGif.css">
    <link rel="stylesheet" href="static/css/modal.css">
    <link rel="stylesheet" href="static/css/modal-content.css">
    <link rel="stylesheet" href="static/css/caption.css">
    <link rel="stylesheet" href="static/css/close.css">
</head>

<style>  /* CSS properties of this page */


    body { /* behaviour of the page body */
        font-family: "Lato", sans-serif;
        transition: background-color .5s;
        margin-top: 0;
        margin-left: 0;
        margin-right: 0;
    }


    #main {
        transition: margin-left .5s;
        padding: 16px;
        margin-right: 15%;
        padding-left: 5%;
        text-align: justify
    }

    #scrollDiv p a {
        font-size: 12px;
        color: white;
        text-decoration: none
    }

    #main a {

        padding-top: 20%;
        margin-top: 20%;
    }

    .table {
        border: 0;
        width: 100%;
        max-width: 100%;
    }

    .tr {
        border: 0;
        width: 100%;
        max-width: 100%;
        border: #057c62;
        border-bottom: #057c62;
        padding-bottom: 10px;
    }

    .th-title {
        text-align: center;
        width: 14.29%;
        font-size: 14px;
        border: #057c62;
        border-bottom: #057c62;
    }
    .th-body {
        border: 0;
        text-align: center;
        width: 14.29%;
        font-size: 12px;
    }

    #main h1 {

        padding-top: 20px;

    }

        #main h3 {

        padding-top: 15px;

    }


</style>

<body>

    <!-- Initiate side navigation bar -->
    <div id="mySidenav" class="sidenav">
        <a href="javascript:void(0)" class="closebtn" onclick="closeNav()">&times; </a>

        <!-- An initial padding-top for avoiding an overlay with the close button -->
        <div style="padding-top: 60px"></div>

        <!-- Initiate an accordion button representing a chapter of the tutorial -->
        <button class="accordion"><i>merlin</i> overview</button>
            <!-- Upon clicking the accordion button a panel is revealed with multiple hyperlinks representing the sub chapters associated with a given chapter -->
            <div class="panel">
                <!-- Initiate a hyperlink representing a subchapter-->
                <!-- The # symbol links the subchapter name to the corresponding section of the myText divider with the same name-->
                <a href="tutorialMainPage.html#overview">Overview</a>
            </div>
        <!-- -->

        <button class="accordion">Before starting</button>
            <div class="panel">
                <a href="tutorialMainPage.html#recommendations">Recommendations</a>
                <a href="tutorialMainPage.html#openMerlin">Open merlin</a>
                <a href="tutorialMainPage.html#selectDatabase">Select database type and set user credentials</a>
            </div>


        <button class="accordion">Workspace</button>
            <div class="panel">
                <a href="tutorialMainPage.html#create">Create a workspace</a>
                <a href="tutorialMainPage.html#load">Load a workspace</a>
                <a href="tutorialMainPage.html#import">Import a workspace</a>
                <a href="tutorialMainPage.html#export">Export a workspace</a>
                <a href="tutorialMainPage.html#clean">Clean a workspace</a>
                <a href="tutorialMainPage.html#delete">Delete a workspace</a>
                <a href="tutorialMainPage.html#clone">Clone a workspace</a>
                <a href="tutorialMainPage.html#compatibility">Compatibility (<i>merlin</i> v3 to v4)</a>
            </div>

        <button class="accordion">Genome files</button>
            <div class="panel">
                <a href="tutorialMainPage.html#genomeFiles">Genome files</a>
                <a href="tutorialMainPage.html#importGenome">Import files</a>
                <a href="tutorialMainPage.html#download">Download <i>NCBI</i> genome files</a>
                <a href="tutorialMainPage.html#exportGenome">Export files</a>
            </div>

        <button class="accordion">Metabolic data</button>
            <div class="panel">
                <a href="tutorialMainPage.html#metabolicData">Metabolic data</a>
            </div>

        <button class="accordion">Genome functional annotation</button>
            <div class="panel">
                <a href="tutorialMainPage.html#gfa">Genome functional annotation</a>
                <a href="tutorialMainPage.html#homology">Homology search</a>
                <a href="tutorialMainPage.html#domain">Domain search</a>
                <a href="tutorialMainPage.html#loadGFA">Load genome functional annotation</a>
                <a href="tutorialMainPage.html#pipelines">Pipelines for performing genome functional annotation</a>
            </div>

            <button class="accordion">Enzymes annotation</button>
        <div class="panel">
            <a href="tutorialMainPage.html#enzymesBoard">Enzymes board</a>
            <a href="tutorialMainPage.html#enzymesAnnotationIntegration">Enzymes annotation integration</a>
        </div>

            <button class="accordion">Transporters annotation</button>
        <div class="panel">
            <a href="tutorialMainPage.html#transportersAnnotation">Transporters annotation</a>
            <a href="tutorialMainPage.html#transyt">Run <i>TranSyT</i></a>
            <a href="tutorialMainPage.html#cleanTransport">Clean transport reactions</a>
        </div>

            <button class="accordion">Compartments annotation</button>
        <div class="panel">
            <a href="tutorialMainPage.html#compartmentsAnnotation">Compartments annotation</a>
            <a href="tutorialMainPage.html#compartmentsBoard">Compartments board</a>
            <a href="tutorialMainPage.html#compartmentsIntegration">Compartments integration</a>
            <a href="tutorialMainPage.html#cleanCompartments">Clean compartments integration</a>
        </div>

            <button class="accordion">Biomass formulation</button>
        <div class="panel">
            <a href="tutorialMainPage.html#biomassFormulation">Biomass formulation</a>
            <a href="tutorialMainPage.html#creatingBiomass">Creating the biomass equation</a>
            <a href="tutorialMainPage.html#calculatingProtein">Calculating protein, <i>DNA</i> and <i>RNA</i> contents</a>
            <a href="tutorialMainPage.html#otherSettings">Other settings</a>
            <a href="tutorialMainPage.html#results">Results</a>
        </div>

            <button class="accordion">Creating drains/exchange reactions</button>
        <div class="panel">
            <a href="tutorialMainPage.html#creatingDrains">Creating drains/exchange reactions</a>
            <a href="tutorialMainPage.html#cleaningDrains">Clean the drains/exchange reactions</a>
        </div>

            <button class="accordion">Creating Gene-Protein-Reaction associations</button>
        <div class="panel">
            <a href="tutorialMainPage.html#creatingGPR">Creating Gene-Protein-Reaction associations</a>
            <a href="tutorialMainPage.html#changeGPR">Change the <i>GPR</i> rules settings</a>
        </div>

            <button class="accordion">Model curation - Boards</button>
        <div class="panel">
            <a href="tutorialMainPage.html#modelCuration">Model curation</a>
            <a href="tutorialMainPage.html#boardsGenes">Genes</a>
            <a href="tutorialMainPage.html#boardsProteins">Proteins</a>
            <a href="tutorialMainPage.html#boardsMetabolites">Metabolites</a>
            <a href="tutorialMainPage.html#boardsReactions">Reactions</a>
            <a href="tutorialMainPage.html#boardsPathways">Pathways</a>

        </div>

            <button class="accordion">Model curation - Tools</button>
        <div class="panel">
            <a href="tutorialMainPage.html#correctRevers">Correct reversibility</a>
            <a href="tutorialMainPage.html#findUnbalanced">Find unbalanced reactions</a>
            <a href="tutorialMainPage.html#findBlocked">Find blocked reactions</a>
            <a href="tutorialMainPage.html#removeBlocked">Remove blocked reactions</a>
            <a href="tutorialMainPage.html#findGene">Find gene</a>
            <a href="tutorialMainPage.html#bioiso"><i>BioISO</i></a>
            <a href="tutorialMainPage.html#runBioiso">Run <i>BioISO</i></a>
            <a href="tutorialMainPage.html#bioisoResults"><i>BioISO</i> results</a>
            <a href="tutorialMainPage.html#pipelineBioiso">A pipeline for performing manual curation with <i>BioISO</i></a>
        </div>

            <button class="accordion">Exporting a model</button>
        <div class="panel">
            <a href="tutorialMainPage.html#exportingModel">Exporting a model</a>
        </div>


            <button class="accordion">Importing a model</button>
        <div class="panel">
            <a href="tutorialMainPage.html#importingModel">Importing a model</a>
        </div>

            <button class="accordion">Plugins manager</button>
        <div class="panel">
            <a href="tutorialMainPage.html#pluginsUpdating">Updating plugins</a>
            <a href="tutorialMainPage.html#pluginsInstalling">Installing new plugins</a>
        </div>

        <button class="accordion">Other tools</button>
        <div class="panel">
            <a href="tutorialMainPage.html#resetLocus">Reset locus tag</a>
            <a href="tutorialMainPage.html#increaseMemory">Increase maximum heap memory size</a>
        </div>

        <!-- A final padding-bottom for avoiding opening panels beyond the window border -->
        <div style="padding-bottom: 70px"></div>
    </div>


        <!-- Scrolling header-->
        <div id="scrollDiv" style="background: #008080; max-width: 100%; padding: 20px">
            <span style="font-size:30px;cursor:pointer;;color:white" onclick="openNav()">&#9776;&nbsp;<i>merlin</i> v4 Tutorial</span>
            <p style="font-size: 12px; color: white">by
                <a href="https://www.ceb.uminho.pt/People/Profile/diogolima">Diogo Lima</a>,
                <a href="https://www.ceb.uminho.pt/People/Profile/davidelagoa">Davide Lagoa</a>,
                <a href="https://www.ceb.uminho.pt/People/Profile/fernandocruz">Fernando Cruz</a>,
                <a href="https://www.ceb.uminho.pt/People/Profile/josebastos">José Bastos</a>,
                <a href="https://www.ceb.uminho.pt/People/Profile/miguelsilva">Miguel Silva</a>,
                <a href="https://www.ceb.uminho.pt/People/Profile/odias">Oscar Dias</a>
                | Oct 20, 2019 | <a href="https://www.ceb.uminho.pt/">CEB Uminho</a>
            </p>
        </div>

        <!-- Main section-->
        <div id="main" onclick="closeNav()">

            <br>

            <h1><i>merlin</i> overview</h1>

                <a name="overview"></a>
                <i>merlin</i> is a simple, graphical and user-oriented solution for the reconstruction of
            genome-scale metabolic models. <i>merlin</i> will guide you along the model reconstruction, providing
            several tools that help to improve and curate the model throughout the whole process.

            <br>
            <br>
            <img class="tutorialGif" src="static/images/MerlinMainWindow.png" alt="merlin overview" id="myImg-1"
                 onclick="modalImage('myImg-1', 'myModal-1', 'img-1', 'caption-1', 'close-1')">
            <div id="myModal-1" class="modal">
                <span class="close" id="close-1">&times;</span>
                    <img class="modal-content" id="img-1">
                        <div id="caption-1" class="caption"> </div>
            </div>
            <br>
            <br>

            <i>merlin</i> main window is divided in several sections. Firstly, the "Operations Menu" includes
            workspace configuration and all the major model reconstruction related operations. The ‘Workspace
            Dashboard’ contains the opened workspaces, as well as the composing dashboards "model", "annotation" and
            "validation". Upon "opening" a selected dashboard, several "boards" become available.

            <br>
            <br>

            Finally, the major part of <i>merlin</i> window is filled by selecting the intended board, in the workspace
            dashboard section. This board is composed by different sections, including:

                <ul style="background:none">
                    <li><b>Board Data Viewer -</b> Display of useful data of the selected board;</li>
                    <li><b>Board Tools -</b> operations related to the current displayed data;</li>
                    <li><b>Board Tabs -</b> These can be used to cycle between the different tabs of a board.</li>
                </ul>


            <h1>Before starting</h1>
                <!-- "name = " declares the name of this section, it is necessary to be able to hyperlink the sidebar to this sub chapter-->

                <a name="recommendations"></a>
                <h3>Recommendations</h3>
                    Before you start we suggest the following optional procedures to enhance the reconstruction of
            Genome-Scale Metabolic Models using <i>merlin</i>:
                    <ul style="background:none">
                        <li><b>BLAST+ -</b> Consider installing NCBI's standalone BLAST+ tool. This tool can be used
                            for hastening the GPRs procedure. Follow the instructions: <a
                                href="https://www.ncbi.nlm.nih.gov/books/NBK52637/" target="_blank"
                                style="padding: 0; margin: 0">Windows</a>; <a
                                href="https://www.ncbi.nlm.nih.gov/books/NBK52640/" target="_blank"
                                style="padding: 0; margin: 0">Unix</a>; <a
                                href="https://www.ncbi.nlm.nih.gov/books/NBK279671/" target="_blank" style="padding:
                                0; margin: 0">Others</a>;</li>
                        <li><b>Memory -</b> Consider increasing
                            the maximum heap memory size (RAM) for large genome sequences or multiple
                            reconstructions. See <a href="tutorialMainPage.html#increaseMemory">Increase maximum heap
                                memory size</a> for more details.</li>
                    </ul>

                <a name="openMerlin"></a>
                <h3>Open merlin</h3>
                    Unzip the merlin folder that you have just downloaded and double-click on the merlin.bat file.
                    <br>
                    <br>
                    <img class="tutorialGif" src="static/images/init.gif" alt="Open merlin" id="myImg0" onclick="modalImage('myImg0', 'myModal0', 'img0', 'caption0', 'close0')">
                    <div id="myModal0" class="modal">
                        <span class="close" id="close0">&times;</span>
                            <img class="modal-content" id="img0">
                                <div id="caption0" class="caption"> </div>
                    </div>

                <a name="selectDatabase"></a>
                <h3>Database settings</h3>
            If the user wants to use the default database settings of <i>merlin</i>, the following instructions can be skipped. Otherwise, to select the database type, the user must open the section “settings” in the menu and then select “database”. Here, the database type is the first option, where the option “H2” corresponds to a local database and is already selected by default. Alternatively, <i>merlin</i> also supports local or remote <i>MySQL</i> databases. The following fields are: username, password, host, and port. When all parameters are set, <i>merlin</i> should be able to successfully establish a connection with the database server of the user.

                <br>
                <br>
                <img class="tutorialGif" src="static/images/db_settings.gif"
                     alt="Database settings"
                     id="myImg37" onclick="modalImage('myImg37', 'myModal37', 'img37', 'caption37', 'close37')">
                    <div id="myModal37" class="modal">
                        <span class="close" id="close37">&times;</span>
                            <img class="modal-content" id="img37">
                                <div id="caption37" class="caption"></div>
                    </div>

            <h1>Workspace</h1>
                <!-- "name = " declares the name of this section, it is necessary to be able to hyperlink the sidebar to this sub chapter-->
                <a name="create"></a>
                <h3>Create a workspace</h3>

                    To create a new workspace, one has to access the “workspace” section and choose the “open” option.
                    If a given user is creating a workspace for the first time, the interface will not provide a list of available workspaces, instead it will prompt users to automatically create a workspace.
                    To create a workspace at any given time, one must choose the “create” option and enter the new workspace name (duplicate names are not possible and will prompt a warning).
            After creating a workspace it is mandatory to enter the <i>NCBI Taxonomy Identifier</i> of the organism in study.
            It is possible to automatically download all <i>NCBI</i> files associated with a given assembly record, as selected by the user, by choosing the “search” option.
            The final step towards creating a new workspace consists in entering the user email. This information will be stored only in the local machine of the user, and it is mandatory to access the <i>APIs</i> of certain frameworks, such as <i>UniProt</i> and <i>BLAST</i>.
                    <br>
                    <br>
                    <img class="tutorialGif" src="static/images/create_ws.gif" alt="Create a workspace" id="myImg" onclick="modalImage('myImg', 'myModal', 'img01', 'caption', 'close')">
                    <div id="myModal" class="modal">
                        <span class="close" id="close">&times;</span>
                            <img class="modal-content" id="img01">
                                <div id="caption" class="caption"> </div>
                    </div>

                <a name="load"></a>
                <h3>Load a workspace </h3>

                    After creating a workspace, the loading process is identical. One must access the “workspace” section and choose the “open” option. Afterwards it is only necessary to choose the desired workspace from a list of available options. The taxonomy identifier and personal email will be already filled.

                    <br>
                    <br>
                    <img class="tutorialGif" src="static/images/load_ws.gif"
                         alt="Load workspace"
                         id="myImg38" onclick="modalImage('myImg38', 'myModal38', 'img38', 'caption38', 'close38')">
                        <div id="myModal38" class="modal">
                            <span class="close" id="close38">&times;</span>
                                <img class="modal-content" id="img38">
                                    <div id="caption38" class="caption"></div>
                        </div>

                <a name="import"></a>
                <h3>Import a workspace </h3>

            <i>merlin</i> provides an option to import pre-existing workspaces, regardless of origin. To import a workspace one must access the “workspace” section and hover over the “import” option, followed by choosing the “workspace” option. A new panel will pop up where users can browse their local machine to select the “.mer” <i>merlin</i> workspace backup. It is possible to redefine a new name for the imported workspace. If the user decides to not provide a new name, the imported workspace will inherit its original name.
                    <br>
                    <br>
                    <img class="tutorialGif" src="static/images/import_ws.gif" alt="Import a workspace" id="myImg2" onclick="modalImage('myImg2', 'myModal2', 'img02', 'caption2', 'close2')">
                    <div id="myModal2" class="modal">
                        <span class="close" id="close2">&times;</span>
                            <img class="modal-content" id="img02">
                                <div id="caption2" class="caption"></div>
                    </div>

                <a name="export"></a>
                <h3>Export a workspace </h3>

            <i>merlin</i> provides an option to export a backup of a given workspace, which can be used for personal storage or sharing with other users. To export a workspace one must access the “workspace” section, hover over the “export” option and choose “workspace”. A new panel will pop up where users can choose which workspace to export, and where to export it in the local machine. Workspaces are exported in “.mer” format and are automatically named based on the workspace name and time/date of export.
                    <br>
                    <br>
                    <img class="tutorialGif" src="static/images/export_ws.gif" alt="Export a workspace" id="myImg3" onclick="modalImage('myImg3', 'myModal3', 'img03', 'caption3', 'close3')">
                        <div id="myModal3" class="modal">
                            <span class="close" id="close3">&times;</span>
                                <img class="modal-content" id="img03">
                                    <div id="caption3" class="caption"></div>
                        </div>

                <a name = "clean"></a>
                <h3>Clean a workspace </h3>

            “Cleaning” a workspace consists in deleting specific parts of the database associated with the workspace, as user-defined. To clean a workspace one must access the “workspace” section and choose the “clean” option. A panel will pop up asking users to choose the target workspace. Additionally, users can choose to clean the whole database of the workspace, model data only, or annotation-data only (including the different annotation stages). Moreover, it is possible to choose to maintain/delete gene sequences and associated data, as well as automatically reloading <i>KEGG’s</i> metabolic data (to the selected workspace after cleaning).
                    <br>
                    <br>
                    <img class="tutorialGif" src="static/images/clean_ws.gif" alt="Clean a workspace" id="myImg4"
                         onclick="modalImage('myImg4', 'myModal4', 'img04', 'caption4', 'close4')">
                        <div id="myModal4" class="modal">
                            <span class="close" id="close4">&times;</span>
                                <img class="modal-content" id="img04">
                                    <div id="caption4" class="caption"></div>
                        </div>


                <a name="delete"></a>
                <h3>Delete a workspace </h3>

                    Deleting a workspace is an irrevocable operation which will also delete the associated database. To delete a workspace one must access the “workspace” section and choose the “delete” option. A panel will pop up asking for the target workspace to delete.

                    <br>
                    <br>
                    <img class="tutorialGif" src="static/images/delete_ws.gif" alt="Delete a workspace" id="myImg5"
                         onclick="modalImage('myImg5', 'myModal5', 'img05', 'caption5', 'close5')">
                        <div id="myModal5" class="modal">
                            <span class="close" id="close5">&times;</span>
                                <img class="modal-content" id="img05">
                                    <div id="caption5" class="caption"></div>
                        </div>

                <a name="clone"></a>
                <h3>Clone a workspace </h3>

            <i>merlin</i> easily allows users to make a copy of available workspaces efficiently without requiring additional export and import operations. To clone a workspace one must access the “workspace” section and click the “clone workspace” section. A new panel will pop up asking users to select the workspace to clone and the name of the generated copy.  If a user enters a name for a copy which matches an already existing workspace, the operation will not continue. If the user does intend to replace an existing workspace with a copy of another workspace, this restriction can be bypassed by choosing the “force database creation” option.

                    <br>
                    <br>
                    <img class="tutorialGif" src="static/images/clone_ws.gif" alt="Clone a workspace" id="myImg6"
                         onclick="modalImage('myImg6', 'myModal6', 'img06', 'caption6', 'close6')">
                        <div id="myModal6" class="modal">
                            <span class="close" id="close6">&times;</span>
                                <img class="modal-content" id="img06">
                                    <div id="caption6" class="caption"></div>
                        </div>

                <a name="compatibility"></a>
                <h3>Compatibility (import workspace from merlin v3 to merlin v4) </h3>

            To import a workspace from <i>merlin</i> 3 to <i>merlin</i> 4, the user must open the operation located at “workspace” -> “compatibility” -> “import workspace from <i>merlin</i> 3”. A window should open where the user must indicate the name of the workspace to be generated in <i>merlin</i> 4. Note that this name should not already exist in the database, otherwise merlin will throw an error claiming that such workspace already exists. If wanted, the user can force <i>merlin</i> to override this database by selecting the option to “force database creation”. The second field is the name of the workspace to be imported from <i>merlin</i> 3. The last field is to indicate the path to <i>merlin</i> 3 home folder.

            <a name="genomeFiles"></a>
            <h1>Genome files </h1>

            Before starting the genome annotation stage it is necessary to provide the genome data to <i>merlin</i>. The genome of the organism in study is a core basis for all the necessary steps towards reconstructing the associated metabolic model. Genome data may be integrated in <i>merlin</i> with local files or by accessing <i>NCBI</i> remotely. Additionally, it is also possible to automate genome files retrieval upon creating a new workspace. After entering the <i>NCBI Taxonomy Identifier</i> of the organism in study, and clicking the “search” option, it is possible to download the genome data from <i>GenBank</i> or <i>RefSeq</i> (and automatically integrate it in <i>merlin</i>)

            <a name="importGenome"></a>
            <h3>Import genome files </h3>

            As aforementioned, it is possible to load genome data to <i>merlin</i> using local files. To perform this operation, one must access the “workspace” section, hover over the “import” option and choose the “genome” option. A new panel will pop up with additional settings. The “search locus tags” option, if activated, triggers an operation to associate each given gene (in the genome files) to a locus tag (if successfully found). merlin is compatible with multiple genome file formats, such as “.faa”, “.fna” and “.gbff”. Upon choosing the genome file to be imported, the final step consists in choosing the desired workspace for the data to be loaded into.

                <br>
                <br>
                <img class="tutorialGif" src="static/images/import_genome_file.gif" alt="Import genome files"
                     id="myImg7" onclick="modalImage('myImg7', 'myModal7', 'img07', 'caption7', 'close7')">
                    <div id="myModal7" class="modal">
                        <span class="close" id="close7">&times;</span>
                            <img class="modal-content" id="img07">
                                <div id="caption7" class="caption"></div>
                    </div>

            <a name="download"></a>
            <h3>Download NCBI genome files </h3>

            If a user did not automatically download the genome files from <i>NCBI</i> during the creation of a new workspace it is possible to do so at any other moment. To perform this operation, one must access the “workspace” section and choose the “download <i>NCBI</i> files” option. If necessary, one might choose the “search” option to filter a specific assembly record from <i>GenBank</i> or <i>RefSeq</i>.

                <br>
                <br>
                <img class="tutorialGif" src="static/images/download_ncbi.gif" alt="Download NCBI genome files"
                     id="myImg8" onclick="modalImage('myImg8', 'myModal8', 'img08', 'caption8', 'close8')">
                    <div id="myModal8" class="modal">
                        <span class="close" id="close8">&times;</span>
                            <img class="modal-content" id="img08">
                                <div id="caption8" class="caption"></div>
                    </div>

            <a name="exportGenome"></a>
            <h3>Export genome files </h3>
                It is possible to export manually or automatically loaded genome files by accessing the “workspace” section, hovering over the “export” option, and choosing the “genome files” option. A new panel will pop up asking the user to choose the genome file to be exported and the output directory.

                <br>
                <br>
                <img class="tutorialGif" src="static/images/export_genome_files.gif" alt="Export a genome files"
                     id="myImg9" onclick="modalImage('myImg9', 'myModal9', 'img09', 'caption9', 'close9')">
                    <div id="myModal9" class="modal">
                        <span class="close" id="close9">&times;</span>
                            <img class="modal-content" id="img09">
                                <div id="caption9" class="caption"></div>
                    </div>

            <a name="metabolicData"></a>
            <h1>Metabolic data </h1>

            Obtaining metabolic information is mandatory for a <i>GSM</i> (genome-scale metabolic) model
            reconstruction. Therefore, <i>merlin</i> contains a simple operation which retrieves information from
            biological databases. Moreover, this operation will load the metabolic information including compounds,
            enzymes, reactions and pathways into <i>merlin’s</i> internal database. Currently, <i>merlin</i> only
            supports metabolic data retrieval from <i>KEGG</i>. However, in the near future, tools to retrieve data
            from <i>BIGG</i> and <i>ModelSEED</i> databases will be available (see <a
                href="https://merlin-sysbio.org/" target="_blank">ongoing work</a> for further information).
            Finally, this
            operation integrates all the retrieved spontaneous reactions into the model, assembling an initial primitive metabolic network, which will be complemented upon integrating the enzymatic and transporters annotation.

                <br>
                <br>
                <img class="tutorialGif" src="static/images/load_kegg.gif" alt="Load KEGG metabolic data"
                     id="myImg10" onclick="modalImage('myImg10', 'myModal10', 'img10', 'caption10', 'close10')">
                    <div id="myModal10" class="modal">
                        <span class="close" id="close10">&times;</span>
                            <img class="modal-content" id="img10">
                                <div id="caption10" class="caption"></div>
                    </div>

            <a name="gfa"></a>
            <h1>Genome functional annotation </h1>

            <i>merlin</i> allows you to perform the genome functional annotation of the organism of interest. This coincides with the first step towards the reconstruction of your <i>GSM</i> model. There are several options for performing the genome functional annotation with <i>merlin</i> involving homology or domain searches to the genome sequence of the organism of interest. Alternatively, one can also load external genome functional annotations.
            In the genome functional, we are going to assign metabolic functions, namely the gene products and enzyme commission (<i>EC</i>) numbers, to each coding sequence (<i>CDS</i>) in the genome.


            <a name="homology"></a>
            <h3>Homology search</h3>

            In <i>merlin</i>, the enzymes functional annotation procedure is leveraged by semi-automatic similarity searches to remote databases, such as <i>UniProtKB</i> or <i>UniProtKB/SwissProt</i>, to find homologous protein sequences.
            The rationale for annotating a given gene or protein based on the similarity of its sequence with other gene or protein sequences relies on the fact that these may have arisen from a common ancestral lineage.
            To perform a homology search in <i>merlin</i>, you must have imported the genome files (see <a href="tutorialMainPage.html#genomeFiles">Genome Files</a> section).

           <h4><i>BLAST</i></h4>
            The Basic Local Alignment Search Tool (<i>BLAST</i>) is the preferred similarity search tool for performing the enzymes functional annotation in <i>merlin</i>. One can easily use the remote web-service from <i>EBI</i> using the default parameters.
Otherwise, you can select the most adequate databases, parameters and thresholds. To do so, you should take a look on your own to the meanings of “expected value” (e-value). Hint: the lower the e-value the highest the similarity score between two sequences.
            Also, you should inspect the available databases:
            <br>
            <br>
            <ul style="background:none">
                <li><b><i>UniProtKB</i> -</b> reviewed and unreviewed annotation records from <i>UniProtKB</i> and <i>SwissProt</i></li>
                <li><b><i>UniProtKB/SwissProt</i> -</b> reviewed annotation records from <i>SwissProt</i></li>
                <li><b><i>UniProtKB/TrEMBL</i> -</b> unreviewed annotation records from <i>UniProtKB</i> </li>
            </ul>
            <br>
            Additionally, <i>merlin</i> supports other <i>UniProtKB</i>-based databases.
            <i>merlin</i> allows you to select the type of the <i>BLAST</i> tool: <i>BLASTp</i> or <i>BLASTx</i>; as well as the sequence type that you are working with. You have the freedom to limit the number of <i>BLAST</i> results to 100 or more. Finally, the substitution matrix can be selected automatically by <i>merlin</i> or defined by you.

            <i>merlin</i> will perform genome-wide sequence alignments automatically and display the results later
            into a simplified layout called enzymes board (see <a href="tutorialMainPage.html#enzymesBoard">Enzymes board</a>).
            This
            process may take several hours depending on the web-server availability, internet connection and genome size.

            <br>
            <br>
            <img class="tutorialGif" src="static/images/ebi_blast.gif" alt="Run EBI BLAST"
                 id="myImg11" onclick="modalImage('myImg11', 'myModal11', 'img11', 'caption11', 'close11')">
                <div id="myModal11" class="modal">
                    <span class="close" id="close11">&times;</span>
                        <img class="modal-content" id="img11">
                            <div id="caption11" class="caption"></div>
                </div>

           <h4>Load <i>BLAST</i> report</h4>

            On the other hand, one can load <i>BLAST</i> reports obtained with the <i>NCBI</i> or <i>EBI</i> web/services. For that, you must provide the path in your computer for the output files returned by these tools and point out their extension, source and type.
            <i>merlin</i> will parse the contents within each <i>BLAST</i> report and load the results into a
            simplified layout called enzymes board (see <a href="tutorialMainPage.html#enzymesBoard">Enzymes board</a>).


            <a name="domain"></a>
            <h3>Domain search</h3>

            <h4><i>InterProScan</i></h4>

            The <i>InterProScan</i> functional annotation is not a mandatory step towards the reconstruction of your genome-scale metabolic model in <i>merlin</i>. In turn, this tool can provide additional information regarding each gene functional annotation.
This tool will provide a functional analysis for each protein sequence by classifying them into families and predicting the presence of domains and important sites.
            <i>InterProScan</i> is a time-consuming tool, thus the user must provide both lower and upper thresholds to limit the number of annotated records.
            The results of your <i>InterProScan</i> functional annotation can be consulted by clicking the magnifying
            glass (coloured with a purple background) of each gene entry in the enzymes board (see <a
                href="tutorialMainPage.html#enzymesBoard">Enzymes board</a>).

            <a name="loadGFA"></a>
            <h3>Load genome functional annotation</h3>

            Although it is highly recommended that you perform your own genome functional annotation to assess with high confidence what are the metabolic capabilities present in the genome of the organism of interest, <i>merlin</i> allows you to skip this step. Instead, one can load external genome functional annotations from <i>KEGG</i> or standard format files, such as <i>GFF3</i> and <i>GenBank</i>.


            <h4><i>KEGG</i> annotation</h4>

            To load a genome functional annotation from <i>KEGG</i>, the user must provide the organism identifier (also known as organism code) in <i>KEGG</i>. For example, "eco" for <i>Escherichia coli</i> K-12 MG1655.
            <i>merlin</i> will load automatically the enzymes functional annotation available in <i>KEGG</i> for the selected organism and integrate it into the model dashboard together with the previously loaded metabolic data (see <a
                href="tutorialMainPage.html#modelCuration">Model curation</a> section).

            <h4><i>GFF3</i></h4>

            To load a genome functional annotation from an external <i>GFF3</i> file, the user must provide the path in your computer for the respective file and its source. Additionally, one must tick the <i>EC</i> option checkbox to allow the integration of metabolic capabilities encoded in the genome (identified by the <i>EC</i> numbers).
            <i>merlin</i> will load the enzymes functional annotation available in the <i>GFF3</i> file and integrate it into the model dashboard together with the previously loaded metabolic data (see <a
                href="tutorialMainPage.html#modelCuration">Model curation</a> section).


            <h4><i>GenBank</i></h4>

            The user can load the genome functional annotation available in the <i>GenBank</i> file downloaded before
            in the <a
                href="tutorialMainPage.html#genomeFiles">Genome files</a> section.
            Alternatively, to load a genome functional annotation from an external <i>GenBank</i> file, the user must provide the path in your computer for the respective file.
            <i>merlin</i> will load the enzymes functional annotation available in the <i>GenBank</i> file and integrate it into the model dashboard together with the previously loaded metabolic data (see <a
                href="tutorialMainPage.html#modelCuration">Model curation</a> section).


            <a name="pipelines"></a>
            <h3>Pipelines for performing genome functional annotation</h3>

            <i>merlin</i> allows you to perform the genome functional annotation following two different pipelines.
            Otherwise, you can perform the genome functional annotation in the enzymes board (see <a
                href="tutorialMainPage.html#enzymesBoard">Enzymes board</a> section).


            <h4>Enzymes annotation - Automatic workflow</h4>

            This tool allows the user to iteratively replicate an annotation workflow. It consists of a <i>GUI</i> containing several slots containing 4 fields to collect parameters from the workflow. Such fields are:<br>
       <ul style="background:none">

           <li>Composed of a drop-down list where the user can select if the step
selected will be relative to the species or the genus. If the empty option is selected,
     the remaining fields become inactive and excluded from the pipeline.</li>
           <li>Composed of a drop-down list containing all species or all genus
               available in <i>merlin’s</i> internal database, depending on the option selected in the first
field. Also the option “any” is available for both options, in order to accept any
     species or any genus.</li>
           <li>Relative to the <i>E-value</i> threshold. Here, the user can define the
     maximum threshold above which all annotations must be disregarded.</li>
           <li>Field is a Boolean option where the user can select between matches with
               <i>UniProt</i> reviewed entries or not.</li>
       </ul>
           <br>
In case that none of the conditions in each slot of the workflow is fulfilled, the user has
available an extra Boolean field that allows the entries to accept the default annotation when
no matches are found.
The process of assessing the correct annotation, for each gene, starts by searching for the best match among the homologous genes found in the similarity search. With this purpose, an entry that meets the requirements of the first slot is sought among all available homologous genes. If no entries match the requirements, the process is repeated for the next slot, until the entire workflow is iterated. If no entries meet the requirements of any slot, the default annotation can be accepted or rejected, depending on the value of the Boolean
option “Accept default annotation if no match is found”.<br><br>
To each annotation, a confidence level is associated depending on which slot of the workflow
the match was found. The confidence level is expressed in sequential order with alphabet letters. The default annotation also has a confidence level of “Default”. When the evaluation process is complete, merlin saves the annotation into its internal database and refreshes the Enzymes Annotation board main tab to reveal the results of the process. The confidence level of each annotation is shown in column “notes”.

            <br>
            <br>
            <img class="tutorialGif" src="static/images/automatic_workflow.gif" alt="Automatic workflow"
                 id="myImg12" onclick="modalImage('myImg12', 'myModal12', 'img12', 'caption12', 'close12')">
                <div id="myModal12" class="modal">
                    <span class="close" id="close12">&times;</span>
                        <img class="modal-content" id="img12">
                            <div id="caption12" class="caption"></div>
                </div>

            <h4><i>SamPler</i></h4>

            <i>SamPler</i> is a semi-automatic method for annotation of enzymes. When <i>merlin’s</i> homology search is complete, a score is assigned to each <i>EC</i> number by weighting the taxonomy score, frequency score and a parameter α. However, the best value for the parameter α is not automatically discovered and has to be calculated manually or using <i>SamPler</i>.
To determine the best value of α, the user has to manually annotate a small sample of
            the genome (5% to 10%), allowing <i>merlin</i> to accept this sample as its standard of truth.<br><br>
            <i>SamPler</i> is available as a plugin at “annotation/enzymes”. When this button is pressed, <i>merlin</i> will launch a warning saying that all previous enzymes annotation will be deleted. At this point, the only options are to cancel the execution or continue. Note also that this tool can only be executed if the <i>BLAST</i> already processed all entries. If the operation was continued, merlin will open a graphical interface containing the sample to annotate. If required, a new sample can be generated using a different size. By pressing the button “save”, the current state of the annotation is saved and it can be continued in a different session. Thus, next time that <i>SamPler</i> is executed, the work already performed is loaded to the graphical interface. During the annotation process, the tool provides a list of <i>EC</i> numbers that can be related to each gene. If the correct option is not available in the list, one should select the option “other ec number”. Likewise, if a gene is not metabolic, the empty option should be selected.<br><br>
            When the annotation of the sample is complete, the button “find best parameters” should be pressed. Here, <i>SamPler</i> starts a wide set of calculations that allow determining the best value of α, upper threshold, and lower threshold. When all calculations are complete, a results graphical interface containing a table is launched. This table allows analyzing the
results attained for each α, namely of the accuracy, upper and lower thresholds, the number of genes above the upper threshold, number of genes below the lower threshold, total of genes for manual curation, the percentage of genes for manual curation and the values the ratio. This window also allows to increase or decrease the values of the precision and negative predictive value.
<br><br>
            The algorithm automatically selects which are the best parameters using the ratio as reference, the higher the ratio, the better the parameters. It is also possible to choose a different option and select other parameters based on the analysis of the results. By pressing the button “apply", the selected parameters are applied to the main window.
Then, at the Enzymes Annotation board main tab, all entries above the upper threshold are accepted as correctly annotated, and below the lower threshold all entries are rejected as non-metabolic genes. All genes between the upper and lower thresholds must then be manually curated.
            <i>SamPler</i> is published and can be checked in detail <a style="padding: 0" href="https://doi.org/10.1186/s12859-019-3038-4">here</a>.

            <br>
            <br>
            <img class="tutorialGif" src="static/images/sampler.gif" alt="SamPler"
                 id="myImg13" onclick="modalImage('myImg13', 'myModal13', 'img13', 'caption13', 'close13')">
                <div id="myModal13" class="modal">
                    <span class="close" id="close13">&times;</span>
                        <img class="modal-content" id="img13">
                            <div id="caption13" class="caption"></div>
                </div>

            <h1>Enzymes annotation</h1>

            <a name="enzymesBoard"></a>
            <h3>Enzymes board</h3>

            The enzymes board shows, to the user, data related to the organism’s genome functional annotation. Its main tab presents all genes, as well as the associated status, name, product, <i>EC</i> number, scores and notes for each one. The status of a gene is divided in two components, the ‘star’, which reflects the gene revision status (Gold - Reviewed; Silver- Unreviewed), and the ‘background’, which reflects if <i>merlin’s</i> annotation is in accordance with <i>UniProt</i> annotation (Green - Annotations in accordance; Light Green - Partial accordance, often happens with promiscuous enzymes; Red - Annotations not in accordance). Additionally, each gene is assigned with a product and <i>EC</i> number based on score calculated by a <i>merlin’s</i> internal algorithm that takes into account both hits frequency and source organism taxonomy. However, products and <i>EC</i> numbers can be manually changed by the user, using the dropdown boxes available in each column. The enzyme board also contains a ‘notes’ column that allows the user to write anything it pretends (Recommendation: the user should take note of any modification made to a gene, as backtracking changes in the model might be necessary). Additional information for each gene can be accessed using the magnifying glass, showing a panel with changeable tabs that can present all hits obtained in the <i>BLAST</i>, protein sequence, <i>BLAST</i> parameters or organism taxonomy. <br>
            <br>
            The search toolbar can be a valuable tool for the user, as it can search by gene name, by notes or in all columns. Setting the search toolbar with ‘all’ allows the user to search for <i>EC</i> numbers in all gene annotations, which can be useful in the later stages of the <i>GSM</i> model reconstruction. The main tab of the enzyme board can also be set to show only results from a selected database. Annotations shown in the board can be exported to an <i>XLSX</i> file or can be used to update the <i>GenBank</i> file of the organism’s genome. Finally, the two ‘buttons’ in the bottom right section of the board are used to save changes made in the board and to integrate the annotations to the model.
    The board also displays a second tab that shows statistical information, such as the number of genes, number of genes without similarities, average number of homologues per gene, among others.

            <br>
            <br>
            <img class="tutorialGif" src="static/images/enzymes_board.gif" alt="Enzymes board"
                 id="myImg14" onclick="modalImage('myImg14', 'myModal14', 'img14', 'caption14', 'close14')">
                <div id="myModal14" class="modal">
                    <span class="close" id="close14">&times;</span>
                        <img class="modal-content" id="img14">
                            <div id="caption14" class="caption"></div>
                </div>

            <a name="enzymesAnnotationIntegration"></a>
            <h3>Enzymes annotation integration</h3>

            <i>merlin</i> easily allows you to integrate the enzymes functional annotation into a draft genome-scale metabolic network. The result of the integration between the loaded metabolic data and the loaded or carried out enzymes functional annotation will define the topology and connectivity of your genome-scale metabolic network.
            The metabolic capabilities encoded in the genome previously identified with <i>EC</i> numbers will now be associated to reactions according to the mappings between enzymes (<i>EC</i> numbers) and reactions available in the metabolic data previously loaded into <i>merlin</i>.
            The following example elucidates what is happening behind the scenes: A given gene has been annotated with the pyruvate formate-lyase <i>EC</i> number (2.3.1.54) in the enzymes annotation step; <i>KEGG</i> metabolic data has been loaded in merlin (see <a
                href="tutorialMainPage.html#metabolicData">Metabolic data</a> section); When this annotation is
            integrated into the model
            dashboard, the following metabolic reactions "R00212" and "R06987" are automatically associated with the referred enzyme and assumed to be present in your genome-scale metabolic model according to the enzymes functional annotation. <i>merlin</i> will perform this operation for all metabolic capabilities (genes assigned with at least one <i>EC</i> number) annotated in the enzymes functional annotation step and according to the loaded metabolic data. <br>
            <br>
            In the Enzymes board, one can perform the integration of the enzymes functional annotation in the model dashboard using the Integrate to model button. This button will directly integrate the results of the enzymes annotation into the model dashboard and fill up the database tables assigned for this module.

            Note that every time the "Integrate to model" button is used the results of the previous integration are lost. This operation saves the new results of the integration on top of the previous ones. Thus, although this button can be used to update the previous integration results, all alterations performed in the model dashboard and database tables designed for this module will be lost.

            The Integrate to model button does not perform the Save operation described in the <a
                href="tutorialMainPage.html#enzymesBoard">Enzymes board</a> section. In other words,
            the enzymes functional annotation can only be saved in the database tables assigned for this purpose using the Save button in the Enzymes board.

            <br>
            <br>
            <img class="tutorialGif" src="static/images/integrate.gif" alt="Integrate enzymes annotation"
                 id="myImg15" onclick="modalImage('myImg15', 'myModal15', 'img15', 'caption15', 'close15')">
                <div id="myModal15" class="modal">
                    <span class="close" id="close15">&times;</span>
                        <img class="modal-content" id="img15">
                            <div id="caption15" class="caption"></div>
                </div>

            <a name="transportersAnnotation"></a>
            <h1>Transporters annotation</h1>


            <i>merlin</i> easily allows you to perform and integrate the transporters’ annotation into a draft genome-scale metabolic network using the state-of-the-art tool called <i>TranSyT</i>. The Transport Systems Tracker (<i>TranSyT</i>) is a new approach to the problem of identifying
            genome-wide transmembrane transport systems, annotating these with reactions. <i>TranSyT</i>
            is the next iteration of <i>TRIAGE</i> (available in <i>merlin</i> v3), though more efficient and designed to overcome its limitations. This new approach still relies on the <i>TCDB</i> to perform the annotation of transporters systems; however, <i>TranSyT</i> automatically retrieves and processes information from this source.

            <a name="transyt"></a>
            <h3>Run <i>TranSyT</i></h3>

            <i>TranSyT</i> is available as a plugin in <i>merlin</i>. By executing this operation, <i>merlin</i> will submit a request to <i>TranSyT’s</i> webserver, in order to generate transport reactions for the organism in study. When the results are retrieved, <i>merlin</i> will automatically integrate those reactions in the model and, if the compartmentalization of the model was already processed, assign compartments to the transport reactions.
            This operation contains some “advanced options” that one can set in order take effect in already compartmentalized models. The fields internal/external/membrane compartment are set as “auto” by default. This means that <i>merlin</i> will try to find the compartments that better suit this localizations. However, if none is found, default compartments will be created for the ones missing. Otherwise, the user can type the name of the compartment to assign to each category, and if these compartments are present in the model, the operation proceeds.
When the process is complete, all reactions generated can be found in the model’s reaction board main tab at “Transporters pathway”.

            <br>
            <br>
            <img class="tutorialGif" src="static/images/transyt.gif"
                 alt="Transporters annotation with TranSyT"
                 id="myImg16" onclick="modalImage('myImg16', 'myModal16', 'img16', 'caption16', 'close16')">
                <div id="myModal16" class="modal">
                    <span class="close" id="close16">&times;</span>
                        <img class="modal-content" id="img16">
                            <div id="caption16" class="caption"></div>
                </div>

            <a name="cleanTransport"></a>
            <h3>Clean transport reactions</h3>

            Cleaning transport reactions generated by <i>TranSyT</i> can be performed by selecting the operation available at: model - remove - transport reactions. This action will remove all <i>TranSyT’s</i> reactions from the model., which will no longer be available at the reactions board.



            <a name="compartmentsAnnotation"></a>
            <h1>Compartments annotation</h1>

            <i>merlin</i> easily allows you to perform and integrate the compartments’ annotation into a draft genome-scale metabolic network using the results of <i>Psortb3</i>, <i>WoLFPSORT</i> and <i>LocTree3</i>.
            To load the reports generated by the aforementioned tools, one must access the “annotation” section, hover over the “compartments” option and choose the “load reports” option. A panel will pop up asking for the respective workspace for the compartment report to be integrated into, and the tool used to generate the report. After clicking “proceed”, it is necessary to enter the URL with the compartment prediction report generated by the aforementioned tools available at the following platforms:<br>
            <ul style="background:none">
                <li><a style="text-decoration: none" href="https://rostlab.org/services/loctree3/"><b><i>LocTree3</i></b></a></li>
                <li><a style="text-decoration: none" href="https://wolfpsort.hgc.jp/"><b><i>WoLFPSORT</i></b></a></li>
                <li><a style="text-decoration: none" href="https://www.psort.org/psortb/"><b><i>Psortb3</i></b></a> - It is worth noting that if a user wishes to integrate a report generated by <i>Psortb3</i>, besides the <i>URL</i> with the results it is also possible to submit a local file with the results’ output format as  “Long Format (tab delimited)”</li>
                </ul>

            <br>
            <br>
            <img class="tutorialGif" src="static/images/compartments.gif"
                 alt="Compartments annotation"
                 id="myImg17" onclick="modalImage('myImg17', 'myModal17', 'img17', 'caption17', 'close17')">
                <div id="myModal17" class="modal">
                    <span class="close" id="close17">&times;</span>
                        <img class="modal-content" id="img17">
                            <div id="caption17" class="caption"></div>
                </div>

            <a name="compartmentsBoard"></a>
            <h3>Compartments board</h3>

            After uploading the reports of either one of the compartments predictors supported by <i>merlin</i>, all predictions should be in this board. Here, results like primary compartment predicted, secondary compartments predicted and respective scores are available. The field “secondary compartments”, containing a drop down box with several different values and set as 0.1 by default, allows the visualization of the secondary compartments with a score difference of less than the selected value. The compartments shown in this table are the ones that will be considered for integration into the model.

            <a name="compartmentsIntegration"></a>
            <h3>Compartments integration</h3>

            In the bottom right corner of the compartments annotation board main tab, it is possible to find the button “integrate to model”. This allows to assign compartments to the reactions in the model. This step is accomplished by checking the enzymes encoded by each gene and the reactions catalyzed by such enzymes. Thus, the reactions associated with such enzymes, will be assigned with the compartments annotated for the encoding genes in the compartments annotation board.
            By pressing this button, a new graphical interface opens. This window contains a table with all compartments predicted to be related with the genes submitted for prediction and an optional feature that lets the user to assign the default compartment membrane. In this table, the user can select which compartments should be ignored by the compartmentalization operation. This feature is useful in cases where some of the compartments predicted do not make sense in the organism in study. For instance, for cases in which the predicted compartment is “unknown”, or even “periplasm” for a gram-positive organism. <br><br>
            Regarding the option to set a default membrane (default membrane when compartmentalizing transport reactions), <i>merlin</i> allows the user to set this property manually or use “auto” mode. In “auto” mode, <i>merlin</i> will automatically try to search for the compartment related with the plasma membrane. If no compartment is found, <i>merlin</i> will launch a warning saying that its search was unsuccessful, and will automatically create a default compartment named “plasma membrane”. When the integration process is complete, the reactions at the model reaction board main tab are expected to become compartmentalized. Note that the column “source” at the reactions board has now switched to “localization”.

            <br>
            <br>
            <img class="tutorialGif" src="static/images/compartments_integration.gif"
                 alt="Integrate compartments annotation"
                 id="myImg18" onclick="modalImage('myImg18', 'myModal18', 'img18', 'caption18', 'close18')">
                <div id="myModal18" class="modal">
                    <span class="close" id="close18">&times;</span>
                        <img class="modal-content" id="img18">
                            <div id="caption18" class="caption"></div>
                </div>

            <a name="cleanCompartments"></a>
            <h3>Clean compartments integration</h3>

            By selecting the button “clean integration” available in the annotation compartments board, all compartments assigned to the reactions in the model will be removed. This will cause the reactions shown in the model reactions board to become non-compartmentalized.


            <a name="biomassFormulation"></a>
            <h1>Biomass formulation</h1>

            <i>merlin</i> easily allows you to create a biomass reaction for the draft genome-scale metabolic network. The formulation of the biomass reaction is one of the most important steps in your reconstruction. This is also the first step towards the conversion of our genome-scale metabolic network into a model. The biomass reaction should account for all the macromolecules that make up the biomass of the organism of interest.
            <br><br>
            <table class="table">
                <tr class="tr">
                    <th class="th-title">Template</th>
                    <th class="th-title">Protein (g/g<sub>DW</sub>)</th>
                    <th class="th-title">DNA (g/g<sub>DW</sub>)</th>
                    <th class="th-title">RNA (g/g<sub>DW</sub>)</th>
                    <th class="th-title">mRNA (g/g<sub>RNA</sub>)</th>
                    <th class="th-title">tRNA (g/g<sub>RNA</sub>)</th>
                    <th class="th-title">rRNA (g/g<sub>RNA</sub>)</th>
                </tr>
                <tr class="tr">
                    <th class="th-title">Archaea</th>
                    <th class="th-body">0.61</th>
                    <th class="th-body">0.034</th>
                    <th class="th-body">0.227</th>
                    <th class="th-body">0.05</th>
                    <th class="th-body">0.15</th>
                    <th class="th-body">0.80</th>
                </tr>
                <tr class="tr">
                    <th class="th-title">Cyanobacteria</th>
                    <th class="th-body">0.595</th>
                    <th class="th-body">0.016</th>
                    <th class="th-body">0.077</th>
                    <th class="th-body">0.05</th>
                    <th class="th-body">0.15</th>
                    <th class="th-body">0.80</th>
                </tr>
                <tr class="tr">
                    <th class="th-title">Gram-positive</th>
                    <th class="th-body">0.588</th>
                    <th class="th-body">0.016</th>
                    <th class="th-body">0.077</th>
                    <th class="th-body">0.05</th>
                    <th class="th-body">0.15</th>
                    <th class="th-body">0.80</th>
                </tr>
                <tr class="tr">
                    <th class="th-title">Gram-negative</th>
                    <th class="th-body">0.533</th>
                    <th class="th-body">0.027</th>
                    <th class="th-body">0.136</th>
                    <th class="th-body">0.05</th>
                    <th class="th-body">0.15</th>
                    <th class="th-body">0.80</th>
                </tr>
                <tr class="tr">
                    <th class="th-title">Mold</th>
                    <th class="th-body">0.367</th>
                    <th class="th-body">0.006</th>
                    <th class="th-body">0.040</th>
                    <th class="th-body">0.05</th>
                    <th class="th-body">0.15</th>
                    <th class="th-body">0.80</th>
                </tr>
                <tr class="tr">
                    <th class="th-title">Yeast</th>
                    <th class="th-body">0.420</th>
                    <th class="th-body">0.026</th>
                    <th class="th-body">0.066</th>
                    <th class="th-body">0.05</th>
                    <th class="th-body">0.15</th>
                    <th class="th-body">0.80</th>
                </tr>
            </table>
            <br>
            <i>merlin</i> provides you with the <i>e-biomass equation tool</i>. This tool allows you to create the biomass reaction automatically including <i>DNA</i>, <i>RNA</i>, Protein, Lipid and Carbohydrate macromolecules as well as the group of cofactors. These metabolites will be denoted as electronic metabolites, namely e-Metabolites. The contents of each e-metabolite in the biomass reaction will be taken from the selected template. Alternatively, these contents can be manually assigned by you. The table below shows the biomass templates available in <i>merlin</i> and the respective contents for each macromolecule.
            <br><br>
            Besides the biomass reaction, the <i>e-biomass equation tool</i> will create a reaction for each main macromolecule (<i>DNA</i>, <i>RNA</i> and Protein) and group of molecules (Cofactors) that makes up the biomass of the organism of interest. According to the reaction, <i>merlin</i> can calculate the contents of each amino acid, deoxyribonucleotide or ribonucleotide from the genome sequence:

            <ul style="background:none">

                <li>Gene translated amino acid sequences, i.e genome protein fasta file (.faa), for the protein assembly</li>
                <li><i>mRNA</i>, i.e. genomic coding sequences fasta file (.fna) as well as <i>tRNA</i> and <i>rRNA</i>, i.e. <i>RNA</i> sequences fasta file (.fna), sequences for the <i>RNA</i> assembly</li>
                <li>Gene nucleotide sequences, i.e. genome sequence fasta file (.fna), for the <i>DNA</i> assembly</li>
            </ul><br>

            Additionally, gene expression data can be used to adjust the amino acids contents. All cofactors in the e-Cofactor reaction are assigned with the same content. Note that, all contents and reactions can be altered manually in the Reactions board (see <a
                href="tutorialMainPage.html#modelCuration">Model curation</a> section). Furthermore, new reactions can
            be created
            for the assembly of other macromolecules in the Reactions board. For instance, the growth-associated energy requirements characterized by the number of ATP molecules required for synthesizing one gram of biomass can be added to the biomass as the hydrolysis of the energy currency of life.


            <a name="creatingBiomass"></a>
            <h3>Creating the biomass equation</h3>

            To set the contents of each macromolecule manually go to the settings sub-menu called tools configurations and click on e-biomass contents. This will determine the fraction of each macromolecule in the biomass reaction.
            Alternatively, you can select in the next steps one of the biomass templates available in <i>merlin</i>.

To create the biomass equation go to the e-biomass equation tool in the model sub-menu called create. If you want to use the manually set contents select the custom template, otherwise select the most adequate template to the organism of interest.

            <br>
            <br>
            <img class="tutorialGif" src="static/images/biomass.gif"
                 alt="e-Biomass tool"
                 id="myImg19" onclick="modalImage('myImg19', 'myModal19', 'img19', 'caption19', 'close19')">
                <div id="myModal19" class="modal">
                    <span class="close" id="close19">&times;</span>
                        <img class="modal-content" id="img19">
                            <div id="caption19" class="caption"></div>
                </div>

            <a name="calculatingProtein"></a>
            <h3>Calculating protein, <i>DNA</i> and <i>RNA</i> contents</h3>

            merlin can calculate automatically the contents of each amino acid, deoxyribonucleotide and
            ribonucleotide into the protein, <i>DNA</i> and <i>RNA</i> reactions, respectively. For that, you must
            have loaded or downloaded all genome files, namely  protein fasta file (.faa), genomic coding sequences
            fasta file (.fna), <i>RNA</i> sequences fasta file (.fna) and genome sequence fasta file (.fna). See <a
                href="tutorialMainPage.html#genomeFiles">Genome files</a> section for more detail.
            Leave the calculate protein, <i>DNA</i> and <i>RNA</i> checkboxes ticked for <i>merlin</i> determining these contents automatically.

            <a name="otherSettings"></a>
            <h3>Other settings</h3>

            The <i>e-biomass equation tool</i> allows you to select the source of the metabolic data used to determine the contents of each amino acid, deoxyribonucleotide and ribonucleotide. Currently, we have support for <i>KEGG</i> and <i>ModelSEED</i> metabolic data.
A biomass compartment can be created specifically for the biomass reactions in the biomass compartment field. Otherwise, leave this field as default for assigning the biomass reactions to the cytoplasmic compartment.
In the advanced options, the user can use gene expression data to adjust each amino acid content in the protein reaction by providing the file path and data separator.

            <br>
            <br>
            <img class="tutorialGif" src="static/images/biomass_manually.gif"
                 alt="e-Biomass tool settings"
                 id="myImg20" onclick="modalImage('myImg20', 'myModal20', 'img20', 'caption20', 'close20')">
                <div id="myModal20" class="modal">
                    <span class="close" id="close20">&times;</span>
                        <img class="modal-content" id="img20">
                            <div id="caption20" class="caption"></div>
                </div>

            <a name="results"></a>
            <h3>Results</h3>

            The results of the <i>e-biomass equation tool</i> will be displayed in a specific pathway called Biomass pathway in the Reactions board (see <a
                href="tutorialMainPage.html#modelCuration">Model curation</a> section).



            <a name="creatingDrains"></a>
            <h1>Creating drains/exchange reactions</h1>

            Exchange reactions or drains are a set of reactions necessary to replicate environmental conditions in which organisms survive and grow, as these can control the metabolite uptake and excretion. <i>merlin</i> provides a tool that automatically creates an exchange reaction for each metabolite in a given compartment. The user may manually insert the compartment for drain creation, if intended, or leave <i>merlin’s</i> default compartment, which is defined as ‘auto’. If the latter is chosen, the software will automatically search for an ‘extracellular’ or ‘outside’ compartment in the model and generate drain reactions for all metabolites in the compartment.
            All drains generated by <i>merlin</i> are grouped into a pathway named ‘Drains Pathway’. Additionally, these reactions contain a single compound as a reactant, no products and its lower and upper bounds are set at a value of 0 and 999999 (default), respectively. These restrictions limit the input and output fluxes of a metabolite and should be constrained according to the environmental condition. For instance, to represent a metabolite available in the media, the lower bound of its drain should be restrained to a negative value. Fluxes determined in experiments may be used to constrain these reactions as well.

            <br>
            <br>
            <img class="tutorialGif" src="static/images/drains.gif"
                 alt="Creating drains"
                 id="myImg21" onclick="modalImage('myImg21', 'myModal21', 'img21', 'caption21', 'close21')">
                <div id="myModal21" class="modal">
                    <span class="close" id="close21">&times;</span>
                        <img class="modal-content" id="img21">
                            <div id="caption21" class="caption"></div>
                </div>

            <a name="cleaningDrains"></a>
            <h3>Clean the drains/exchange reactions</h3>

            As <i>GSM</i> model reconstruction is an iterative process, metabolites available in the model may change throughout the reconstruction. Therefore, <i>merlin</i> provides a tool which allows the user to automatically clean all the drain reactions previously generated by the drain reactions tool.


            <a name="creatingGPR"></a>
            <h1>Creating Gene-Protein-Reaction associations</h1>

            The gene-protein-rules are generated using boolean logic in order to assemble <i>GPR</i> relationships of different complexities. <i>merlin</i> easily allows the user to create <i>GPR</i> associations for the draft genome-scale metabolic network using <i>KEGG Orthology</i> data.
            For each reaction in the model associated with one or more enzymes the algorithm looks for the <i>KEGG</i> orthologous associated with each enzyme. The genes associated with each ortholog are evaluated and a <i>BLAST</i> search against the whole genome is performed with previously defined similarity thresholds . The <i>BLAST</i> similarity thresholds for both reference organism and other organism can be defined in the method ”<i>GPR</i> rules” inside “tools configuration” in the section “settings”. These values allow the algorithm to select the genes that should be selected after the <i>BLAST</i> results have been generated.
            <br><br>
            The enzymes are connected to their orthologs through one or more modules which have a specific definition, comprising different orthologs, depending on the reaction they are associated with. The definition is used to built the boolean rules by using the <i>BLAST</i> evaluation to accept or reject a definition. For example, if in a definition such as “(GeneA OR GeneB)” one of the genes is not selected in the <i>BLAST</i> evaluation procedure the definition is ignored. On the other hand if both genes are accepted the definition is accepted and can be used as one of the  boolean rules for a reaction. The user has the capacity to change these boolean rules in the “edit” button on the reactions view and apply the changes to the model.

            <br>
            <br>
            <img class="tutorialGif" src="static/images/gprs.gif"
                 alt="Creating GPRs"
                 id="myImg22" onclick="modalImage('myImg22', 'myModal22', 'img22', 'caption22', 'close22')">
                <div id="myModal22" class="modal">
                    <span class="close" id="close22">&times;</span>
                        <img class="modal-content" id="img22">
                            <div id="caption22" class="caption"></div>
                </div>

            <a name="changeGPR"></a>
            <h3>Change the <i>GPR</i> rules settings</h3>

            <i>merlin</i> allows you to choose many standards, such as “reference organism similarity threshold”, in order to create <i>GPR</i> associations. To assign different values to the thresholds you can access the GPR rules settings in “tools configuration” in section “settings”.


            <a name="modelCuration"></a>
            <h1>Model curation</h1>

            <i>merlin</i> has been designed to foster the curation of draft genome-scale metabolic networks. For that, <i>merlin</i> provides diverse tools allowing you to curate your model in a semi-automated fashion.

            <a name="boardsGenes"></a>
            <h3>Genes</h3>

            The “genes” board main tab lists all genes including its locus tags, names, number of encoding subunits, number of encoded proteins and number of associated reactions. If one seeks further information regarding a given gene it is possible to click on the magnifying glass associated with an entry to access synonyms, orthologs, encoded proteins, associated compartments and sequences (DNA and Protein). Furthermore, this board contains multiple filters and operations below the gene table. The “search” field eases gene filtering by name or locus tag, highlighting the matched gene (the arrows and indexes on the rightmost position allow cycling between multiple matches). In addition, it is possible to filter genes shown in the board by encoding genes only.
            The “insert” option triggers a new panel which allows users to create a new gene entry, while the “edit” option triggers an operation to modify pre existing genes. The “remove” option removes a given gene both from the board and from the database. The “export file” option triggers an operation to export listed data for each gene in the board in <i>XLSX</i> format. Similarly to other boards, the “genes” board has a statistics tab summarizing essential data, such as the total number of genes, number of protein encoding genes and number of enzyme encoding genes.

            <br>
            <br>
            <img class="tutorialGif" src="static/images/genes_board.gif"
                 alt="Genes board"
                 id="myImg23" onclick="modalImage('myImg23', 'myModal23', 'img23', 'caption23', 'close23')">
                <div id="myModal23" class="modal">
                    <span class="close" id="close23">&times;</span>
                        <img class="modal-content" id="img23">
                            <div id="caption23" class="caption"></div>
                </div>

            <a name="boardsProteins"></a>
            <h3>Proteins</h3>

            The “proteins” board main tab lists all proteins including, names, identifiers, number of involved reactions and number of encoding genes. By clicking on the magnifying glass associated with a given protein, it is possible to visualize additional information such as the encoded reactions of the protein, encoded genes, Gene-Protein-Reaction rules (if available), associated pathways, protein synonyms and associated compartments. The search field allows users to filter data by protein name or identifier. The “insert” and “edit” operations are responsible for creating new proteins and editing pre existing proteins respectively. The “remove” operation removes a chosen protein both from the board and the database. In addition, It is possible to filter shown proteins by “in model” proteins, which are proteins integrated in the model.
Moreover, this board contains an additional statistics tab describing data, such as the number of proteins, number of enzymes and number of transporter proteins in the database.

            <br>
            <br>
            <img class="tutorialGif" src="static/images/protein_board.gif"
                 alt="Protein board"
                 id="myImg24" onclick="modalImage('myImg24', 'myModal24', 'img24', 'caption24', 'close24')">
                <div id="myModal24" class="modal">
                    <span class="close" id="close24">&times;</span>
                        <img class="modal-content" id="img24">
                            <div id="caption24" class="caption"></div>
                </div>

            <a name="boardsMetabolites"></a>
            <h3>Metabolites</h3>

            The “metabolites” board main tab lists all metabolites including names, associated compartments, formulae, external identifiers, number of associated biochemical and transport reactions in the model. By clicking on the magnifying glass associated with a given metabolite, it is possible to visualize additional information such as involved reactions, metabolite synonyms and entry types. The search field allows users to filter metabolites by name, formula and external identifier. The “insert”, “edit” and “remove” operations are responsible for creating new metabolites, editing pre existing ones, and removing a chosen metabolite respectively. Additionally, it is possible to filter the “metabolites” board by reactants, highlighting the respective entries in green, and by products, highlighting the respective entries in blue. All other entries are simultaneously reactants and products. As the “proteins” board , it is possible to filter the “metabolites” board by metabolites integrated in the model. Finally, it is possible to filter metabolite entries by type, including glycans, drugs and compounds. This board contains an additional statistics tab describing data, such as the number of reactants, number of products, number of reactants which are both reactants and products and number number of associated reactions.

            <br>
            <br>
            <img class="tutorialGif" src="static/images/metabolites_board.gif"
                 alt="Metabolites board"
                 id="myImg25" onclick="modalImage('myImg25', 'myModal25', 'img25', 'caption25', 'close25')">
                <div id="myModal25" class="modal">
                    <span class="close" id="close25">&times;</span>
                        <img class="modal-content" id="img25">
                            <div id="caption25" class="caption"></div>
                </div>

            <a name="boardsReactions"></a>
            <h3>Reactions</h3>

            The “reactions” board main tab lists all reactions including the equations, reaction identifiers, associated pathways and sources. The “source” column is an important feature identifying manually created/edited reactions as “MANUAL”, reactions obtained from homology <i>BLAST</i> search results as “HOMOLOGY” and unprocessed reactions retrieved from <i>KEGG</i> as “KEGG”. Additionally, each reaction is identified as reversible (based on upper and lower flux bounds) and as integrated in the model or not. The magnifying glass next to each reaction triggers the appearance of a panel with additional data such the metabolites involved in the reaction, associated enzymes, reaction properties and synonyms, associated pathways and source. <br>
            <br>
            The search field allows users to filter reactions by reaction identifier and equation. Additionally, it is possible to filter the set of shown reactions by associated pathway. Furthermore, after choosing a specific pathway, by clicking in the “draw in browser” option it is possible to access an external page from <i>KEGG</i> with a visual representation of the pathway as represented in <i>merlin</i>. The “insert”, “edit” and “remove” operations are responsible for creating new reactions, modifying pre existing reactions, and removing a chosen reaction respectively. The “find gene” button triggers an operation to find an orthologous gene for a given reaction. The “duplicate” button creates a copy of a chosen reaction, with the only change being the incremented reaction identifier (to distinguish copies from the original reaction). <br>
            <br>
            The pathway subsection contains the “new” and “remove” buttons which trigger operations to create a new pathway or remove a pre existing one. All reactions listed in the “reactions” board are color coded by pathway, thus every reaction belonging to the same pathway has the same color. The “colors” button automatically refreshes the color palette of the “reactions” board , which might increase text legibility. The “export file” button generates a <i>XLSX</i> file with all reactions data, which can be further filtered by pathway. Finally, it is possible to filter the “reactions” board to list only the reactions which are integrated in the model. This board contains an additional statistics tab describing data, such as the number of reactions in the model filtered by source and number of reversible and irreversible reactions.

            <br>
            <br>
            <img class="tutorialGif" src="static/images/reactions_board.gif"
                 alt="Reactions board"
                 id="myImg26" onclick="modalImage('myImg26', 'myModal26', 'img26', 'caption26', 'close26')">
                <div id="myModal26" class="modal">
                    <span class="close" id="close26">&times;</span>
                        <img class="modal-content" id="img26">
                            <div id="caption26" class="caption"></div>
                </div>

            <a name="boardsPathways"></a>
            <h3>Pathways</h3>

            The “pathways” board main tab lists all pathways including its name, number of associated reactions and enzymes and external <i>KEGG</i> identifier. The magnifying glass next to each pathway triggers the appearance of a panel listing all reactions and enzymes associated to a given pathway. The search field allows users to filter pathways by name or external <i>KEGG</i> identifier. The “export file” button generates a <i>XLSX</i> file with all pathway data. The statistics tab of the “pathways” board displays the number of pathways with no SBML file associated.

            <br>
            <br>
            <img class="tutorialGif" src="static/images/pathways_board.gif"
                 alt="Pathways board"
                 id="myImg27" onclick="modalImage('myImg27', 'myModal27', 'img27', 'caption27', 'close27')">
                <div id="myModal27" class="modal">
                    <span class="close" id="close27">&times;</span>
                        <img class="modal-content" id="img27">
                            <div id="caption27" class="caption"></div>
                </div>

            <h1>Tools</h1>

            <a name="correctRevers"></a>
            <h3>Correct reversibility</h3>

            Incorrect reversibility and directionality of reactions in a <i>GSM</i> model can lead to futile cycles, which impairs model’s predictions. Additionally, all reactions retrieved from <i>KEGG</i> display a reversible status, thus its correction is mandatory. <i>merlin</i> offers a tool which corrects reactions (loaded in the database) directionality and reversibility using <i>Zeng</i> or <i>ModelSEED</i> databases as a source of corrections. This tool also contains a user selected parameter referent to the template organism ‘class’ and a force correction option that should be checked so reactions are automatically corrected.

            <br>
            <br>
            <img class="tutorialGif" src="static/images/correct_reversibility.gif"
                 alt="Correct reversibility tool"
                 id="myImg28" onclick="modalImage('myImg28', 'myModal28', 'img28', 'caption28', 'close28')">
                <div id="myModal28" class="modal">
                    <span class="close" id="close28">&times;</span>
                        <img class="modal-content" id="img28">
                            <div id="caption28" class="caption"></div>
                </div>

            <a name="findUnbalanced"></a>
            <h3>Find unbalanced reactions</h3>

            Unbalanced reactions in <i>GSM</i> models can lead to incorrect flux distributions during simulations. Hence, reactions’ stoichiometry has to be ensured, by guaranteeing that the atoms’ sum in reactions’ participating compounds is equal on both sides of the reaction equation.
            As in previous cases, <i>merlin</i> contains a tool that identifies unbalanced reactions. This operation requires a ‘proton name’ input (default: H+), which can be defined by the user. In order to easily pinpoint unbalanced reactions, the tool sets the unbalanced reactions’ names in bold and italic. Additionally, for each reaction in the model, a new tab, named ‘balance’, becomes available in the dropdown box of the reaction data panel, accessed by using the ‘magnifying glass’. Unbalanced reactions can also be exported in an <i>XLSX</i> file.
    The typical unbalanced reaction issues found within a metabolic network include:<br>

            <ul style="background:none">
                <li>Metabolite missing formulae</li>
                <li>Missing/excess protons or water</li>
                <li>Macromolecule synthesis/breakdown</li>
            </ul>

            Usually, the first and second cases are easily solvable by determining the missing formulae and adding/removing protons/water, respectively, whereas the third situation requires the removal of the macromolecule on one side of the reaction equation.
            However, as no specific rules to correct these issues exist, one should always check biological databases such as <i>MetaCyc</i> and <i>BRENDA</i>, to obtain additional information, before correcting each unbalanced reaction.

            <br>
            <br>
            <img class="tutorialGif" src="static/images/find_unbalanced.gif"
                 alt="Find unbalanced reactions tool"
                 id="myImg29" onclick="modalImage('myImg29', 'myModal29', 'img29', 'caption29', 'close29')">
                <div id="myModal29" class="modal">
                    <span class="close" id="close29">&times;</span>
                        <img class="modal-content" id="img29">
                            <div id="caption29" class="caption"></div>
                </div>

            <a name="findBlocked"></a>
            <h3>Find blocked reactions</h3>

            In some occasions, intervenients of reactions are neither consumed or produced by other reactions, which results in a ‘blocked’ reaction. In these situations, flux through the reaction during simulations is impaired, and, in some cases, could affect the model’s performance. Usually, blocked reactions are a result of errors in the enzyme annotation or compartmentalization of the model. In order to aid the analysis of unconnected reactions, <i>merlin</i> has an operation that aims at identifying these reactions, coloring them in red (blocked reaction) or light blue (leads to a block reaction). Additionally, in the reaction data panel, a new tab becomes available, stating which metabolites of the reaction are considered dead-end metabolites (metabolite that cannot be consumed or produced).
            Therefore, the ‘find blocked reactions’ tool is a valuable asset to curate the <i>GSM</i> model, as it can help in the identification of metabolic gaps (missing reactions in a given pathway), correction of enzyme annotations and incorrect directionality/reversibility of reactions.

            <br>
            <br>
            <img class="tutorialGif" src="static/images/find_blocked.gif"
                 alt="Find blocked reactions tool"
                 id="myImg30" onclick="modalImage('myImg30', 'myModal30', 'img30', 'caption30', 'close30')">
                <div id="myModal30" class="modal">
                    <span class="close" id="close30">&times;</span>
                        <img class="modal-content" id="img30">
                            <div id="caption30" class="caption"></div>
                </div>

            <a name="removeBlocked"></a>
            <h3>Remove blocked reactions</h3>

            Usually, the majority of unconnected reactions are unnecessary for a <i>GSM</i> model. Therefore, <i>merlin</i> offers the possibility to remove automatically all the detected blocked reactions using the ‘remove blocked reactions’ operation. However, before using said tool, it is mandatory to identify the blocked reactions firstly, or else the operation launches a warning.


            <a name="findGene"></a>
            <h3>Find gene</h3>

            Ideally, all metabolic reactions in the model should be associated to a gene. "Find Gene" is a tool that identifies gene or genes for a reaction. If an enzyme is associated to a reaction this tool is able to look for orthologs for the gene associated with the enzyme. Based on <i>BLAST</i> similarity thresholds, very similar process occurs on “Create <i>GPR</i>” tool, the tool will filter all the orthologs found and will select only the ones that match the thresholds. The tool will present to the user a panel with the orthologs found together with information from the <i>BLAST</i> performed for each ortholog. Then the user can choose which genes should be added to the model.

            <br>
            <br>
            <img class="tutorialGif" src="static/images/find_gene.gif"
                 alt="Find gene tool"
                 id="myImg31" onclick="modalImage('myImg31', 'myModal31', 'img31', 'caption31', 'close31')">
                <div id="myModal31" class="modal">
                    <span class="close" id="close31">&times;</span>
                        <img class="modal-content" id="img31">
                            <div id="caption31" class="caption"></div>
                </div>

            <a name="bioiso"></a>
            <h3><i>BioISO</i></h3>

            <i>BioISO</i> is aimed at evaluating your biomass formulation or genome-scale metabolic network. This tool is based on the Constraints-Based Reconstruction and Analysis (<i>COBRA</i>) and Flux Balance Analysis (<i>FBA</i>) frameworks.
            <i>BioISO</i> can receive as input the current genome-scale metabolic model and return potential errors in the biomass formulation or gaps in the metabolic network.

            <a name="runBioiso"></a>
            <h3>Run <i>BioISO</i></h3>

            To run <i>BioISO</i>, go to the validation menu and select a reaction to be tested and the objective direction (maximize or minimize the reaction). All reactions available in the model will be displayed in the dropdown box. Leave the <i>URL</i> field as default. It is expected <i>BioISO</i> to take a few minutes (2-5 minutes) depending on the model size, submitted reaction and server availability.

            <a name="bioisoResults"></a>
            <h3><i>BioISO</i> results</h3>

The results will be available under the validation dashboard on the left. One can open the results board identified by the submitted reaction. The results will be displayed in a tabular format by clicking on the corresponding board.
            The precursors (reactants) and products of the submitted reaction are displayed together with the analysis, number of associated reactions and role of the metabolite in the submitted reaction. The evaluation performed by <i>BioISO</i> for each metabolite is based on the analysis of the metabolic network capability to synthesize or consume that given metabolite, depending on the role.
Then, you can take a look at the reactions associated with each metabolite by opening a new panel on the magnifying glass. In this panel, the evaluation result depends on whether the reaction can carry any flux towards the consumption or synthesis of the previous metabolite, depending on the role. The number of reactants and products is also available in this panel.
To get more details about a given reaction, one can open a new panel by clicking on the associated magnifying glass. This panel will display BioISO’s analysis over the metabolites involved in that specific reaction. Finally, you can repeat the last two operations to get more detail over that metabolites and reactions.

            <br>
            <br>
            <img class="tutorialGif" src="static/images/bioISO.gif"
                 alt="BioISO tool"
                 id="myImg32" onclick="modalImage('myImg32', 'myModal32', 'img32', 'caption32', 'close32')">
                <div id="myModal32" class="modal">
                    <span class="close" id="close32">&times;</span>
                        <img class="modal-content" id="img32">
                            <div id="caption32" class="caption"></div>
                </div>

            <a name="pipelineBioiso"></a>
            <h3>A pipeline for performing manual curation with <i>BioISO</i></h3>

            <i>merlin</i> easily allows you to curate your metabolic model. For that, we suggest a straightforward pipeline using <i>BioISO</i>. Once the user has assembled the draft genome-scale metabolic network, performed the reversibility correction, and found both unbalanced and blocked reactions, we suggest the utilization of the <i>BioISO</i> tool to hastening the model curation and debugging.
            To run <i>BioISO</i> one must provide a reaction that will be interpreted as the objective. For instance, the main objective of microorganisms living in wild-type environmental conditions is to grow namely maximizing the biomass production. Following this principle, the first reaction to ever be submitted to <i>BioISO</i> can be the biomass reaction. <i>BioISO</i> will evaluate whether the current model is capable or not to synthesize or consume the metabolites associated with this reaction. If such turns out to be the case, this means that the current model is correctly assembled and thus attaining the biomass production.<br><br>

            In addition to this preliminary analysis, <i>BioISO</i> will conduct its analysis to the reactions associated to these metabolites and so on.
In this way, one can solve potential errors in the draft genome-scale metabolic model by focusing specifically in modules that cannot carry any flux towards the submitted objective. Note that such modules are just a given set of metabolites and reactions that are not correctly assembled and thus the current model is not capable of succeeding in that objective.
            The manual curation process is therefore leveraged with this pipeline, as you can submit all reactions as objectives. If one is interested in evaluating whether a given pathway can synthesize or consume a given metabolite, the corresponding reaction can be submitted for analysis. Then, you would have to take care of the faulty modules (reactions or metabolites), according to <i>BioISO</i> results.
            Finally, you should find all potential errors in the model, correct them and run <i>BioISO</i> for the same objective to see if the changes had an effect on the model.


            <a name="exportingModel"></a>
            <h1>Exporting a model</h1>

            <i>merlin</i> easily allows a user to export a given model in <i>SBML</i> level 2 or 3, a format based on <i>XML</i> used to represent computational models of biological phenomena. To export a model one must access the “workspace” section, hover over the “export” option and choose the “model” option. A new panel will pop up asking for the workspace which contains the model to be exported. Additionally, it is possible to export a text file containing the formulae of model metabolites by selecting the “Generate Formulae” option. After choosing the output directory and name, it is possible to export the model in <i>SBML</i> format.

            <br>
            <br>
            <img class="tutorialGif" src="static/images/export_model.gif"
                 alt="Exporting a model"
                 id="myImg33" onclick="modalImage('myImg33', 'myModal33', 'img33', 'caption33', 'close33')">
                <div id="myModal33" class="modal">
                    <span class="close" id="close33">&times;</span>
                        <img class="modal-content" id="img33">
                            <div id="caption33" class="caption"></div>
                </div>

            <a name="importingModel"></a>
            <h1>Importing a model</h1>

            It is possible to import a pre existing model to <i>merlin</i> as long as it is available in <i>SBML</i> level 2 or 3 format. To perform this operation one must access the “workspace” section, hover over the “import” option and choose the “model” option. A new panel will pop up asking to select the workspace which will host the model to be loaded. After choosing the correct <i>SBML</i> level of the model, a user only needs to choose the actual model file.

            <br>
            <br>
            <img class="tutorialGif" src="static/images/import_model.gif"
                 alt="Importing a model"
                 id="myImg34" onclick="modalImage('myImg34', 'myModal34', 'img34', 'caption34', 'close34')">
                <div id="myModal34" class="modal">
                    <span class="close" id="close34">&times;</span>
                        <img class="modal-content" id="img34">
                            <div id="caption34" class="caption"></div>
                </div>

            <h1>Plugins manager</h1>

            <a name="pluginsUpdating"></a>
            <h3>Updating plugins</h3>

            When <i>merlin</i> is executed, it will automatically search for updates in the plugins installed, and if any updates are found, a pop-up is launched asking to install the updates. If the updates are installed, merlin will need to restart in order for the changes to take effect.
To search for updates manually, an operation is available in the menu on the “settings” option.

            <br>
            <br>
            <img class="tutorialGif" src="static/images/update_plugins.gif"
                 alt="Updating plugins"
                 id="myImg35" onclick="modalImage('myImg35', 'myModal35', 'img35', 'caption35', 'close35')">
                <div id="myModal35" class="modal">
                    <span class="close" id="close35">&times;</span>
                        <img class="modal-content" id="img35">
                            <div id="caption35" class="caption"></div>
                </div>

            <a name="pluginsInstalling"></a>
            <h3>Installing new plugins</h3>

            The operation “repository manager” allows the user to enable, disable and/or install new plugins into <i>merlin</i>. The upper table inside the manager graphical interface allows to verify the plugins that are already installed in merlin as well as other information, and a checkbox to enable or disable each plugin. The lower table contains the plugins that are available in the online repository. To require the installation of a plugin, the button “Install - (...)” regarding the chosen plugin must be pressed inside in the column “Download”. Any modification regarding the plugins in this board requires restarting merlin in order to take effect.

            <br>
            <br>
            <img class="tutorialGif" src="static/images/install_plugins.gif"
                 alt="Installing plugins"
                 id="myImg36" onclick="modalImage('myImg36', 'myModal36', 'img36', 'caption36', 'close36')">
                <div id="myModal36" class="modal">
                    <span class="close" id="close36">&times;</span>
                        <img class="modal-content" id="img36">
                            <div id="caption36" class="caption"></div>
                </div>

            <h1>Other tools</h1>

            <a name="resetLocus"></a>
            <h3>Reset locus tag</h3>

            After performing <i>BLAST</i>, genes’ identifiers might be altered and by integrating the enzymes annotation to the model, genes’ locus tags are changed. Therefore, this operation resets the identifiers to the original genes’ locus tags, using the local download genome files or by performing an online search.

            <a name="increaseMemory"></a>
            <h3>Increase maximum heap memory size</h3>

            It is recommend to increase the maximum heap memory size (RAM) for large genome sequences or multiple
            reconstructions. If the case, one can go to "settings" section in the menu and select "increase memory".
            While, the maximum heap memory size is set by default to 1GB, you can select a higher amount of maximum
            heap memory size in the dropdown menu.
        </div>


    <script src="static/js/scroll.js"></script>
    <script src="static/js/accordion.js"></script>
    <script src="static/js/sidenav.js"></script>
    <script src="static/js/modalImage.js"></script>

</body>
